### PR TITLE
refactor(CAS proxy): multiple CAS backend providers support 

### PR DESCRIPTION
**Pull request by migmartri**
Adds support to dynamically select a backend provider in the CAS proxy

image

- [CAS] Loads a list of backend providers at boot
- [CAS] Selects the CAS backend dynamically in each request
- [Controlplane] Generate a CAS token containing also the backend type, which can be used for the selection above.

This is a prerequisite of #357, once merged we will be able to implement a new backend in CAS + enable it in the controlplane

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -71,4 +71,8 @@ severity:
     - linters:
         - staticcheck
       text: "SA1019:"
-      severity: info
\ No newline at end of file
+      severity: info
+issues:
+  exclude-rules:
+    - path: _test\.go
+      text: "Potential hardcoded credentials"
```
Comment (migmartri) [inline .golangci.yml:78]: for some reason now it complains about some of the sha256 digests that we are using during testing

File: app/artifact-cas/api/cas/v1/status.proto [hunk 2]
```diff
@@ -17,34 +17,31 @@ syntax = "proto3";
...
+  }
 }
 
-message InfozRequest { }
+message InfozRequest {}
 
 message InfozResponse {
-	string version = 1;
+  string version = 1;
+  repeated string backends = 2;
```
Comment (migmartri) [inline app/artifact-cas/api/cas/v1/status.proto:37]: this is in fact the new part. It exposes the enabled backends.

File: app/artifact-cas/cmd/main.go [hunk 3]
```diff
@@ -56,19 +58,36 @@ func init() {
...
-		kratos.Server(
-			gs,
-			hs,
-			ms,
+type app struct {
+	*kratos.App
+	backend.Providers
+}
+
+func loadCASBackendProviders(creader credentials.Reader) backend.Providers {
```
Comment (migmartri) [inline app/artifact-cas/cmd/main.go:66]: the new dynamic loader

**Review by danlishka — APPROVED** [review]
LGTM based on the review via conf call

**Pull request merged**